integrations/test_lightning.py

```diff
@@ -201,45 +201,84 @@ def _assert_called(model, stage):
     _assert_called(model, "test")
 
 
-# todo: reconsider if it make sense to keep here
-# def test_metric_lightning_log(tmpdir):
-#     """ Test logging a metric object and that the metric state gets reset after each epoch."""
-#     class TestModel(BoringModel):
-#         def __init__(self):
-#             super().__init__()
-#             self.metric_step = SumMetric()
-#             self.metric_epoch = SumMetric()
-#             self.sum = 0.0
-#
-#         def on_epoch_start(self):
-#             self.sum = 0.0
-#
-#         def training_step(self, batch, batch_idx):
-#             x = batch
-#             self.metric_step(x.sum())
-#             self.sum += x.sum()
-#             self.log("sum_step", self.metric_step, on_epoch=True, on_step=False)
-#             return {'loss': self.step(x), 'data': x}
-#
-#         def training_epoch_end(self, outs):
-#             self.log("sum_epoch", self.metric_epoch(torch.stack([o['data'] for o in outs]).sum()))
-#
-#     model = TestModel()
-#     model.val_dataloader = None
-#
-#     trainer = Trainer(
-#         default_root_dir=tmpdir,
-#         limit_train_batches=2,
-#         limit_val_batches=2,
-#         max_epochs=2,
-#         log_every_n_steps=1,
-#         weights_summary=None,
-#     )
-#     trainer.fit(model)
-#
-#     logged = trainer.logged_metrics
-#     assert torch.allclose(tensor(logged["sum_step"]), model.sum)
-#     assert torch.allclose(tensor(logged["sum_epoch"]), model.sum)
+def test_metric_lightning_log(tmpdir):
+    """Test logging a metric object and that the metric state gets reset after each epoch."""
+
+    class TestModel(BoringModel):
+        def __init__(self):
+            super().__init__()
+            self.metric_step = SumMetric()
+            self.metric_epoch = SumMetric()
+            self.sum = 0.0
+
+        def on_epoch_start(self):
+            self.sum = 0.0
+
+        def training_step(self, batch, batch_idx):
+            x = batch
+            self.metric_step(x.sum())
+            self.sum += x.sum()
+            self.log("sum_step", self.metric_step, on_epoch=True, on_step=False)
+            return {"loss": self.step(x), "data": x}
+
+        def training_epoch_end(self, outs):
+            self.log("sum_epoch", self.metric_epoch(torch.stack([o["data"] for o in outs]).sum()))
+
+    model = TestModel()
+    model.val_dataloader = None
+
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        limit_train_batches=2,
+        limit_val_batches=2,
+        max_epochs=2,
+        log_every_n_steps=1,
+        weights_summary=None,
```

Comment:
Do `metrics` tests not run with pl_master?
I think this flag is deprecated.

Reply:
```suggestion
```